File: rtrim/rtrim.js
// ATOMIC_FUNCTION:  rtrim($val)
// LANG:  node.js
// FUNCTION: rtrim($val) 
function rtrim(str, chr) {
    var rgxtrim = (!chr) ? new RegExp('\\s+$') : new RegExp(chr+'+$');
    return str.replace(rgxtrim, '');
}
var start_val = "  abcdefg  ";
var expected_end_val = "  abcdefg";
var end_val = rtrim(start_val);

// TEST
if(end_val == expected_end_val) {
	console.log("PASS");
} else {
	console.log("FAIL");
}

File: rtrim/rtrim.java
import java.util.regex.Pattern;

public class rtrim {
    private final static Pattern RTRIM = Pattern.compile("\\s+$");

    public static String rtrim(String s) {
        return RTRIM.matcher(s).replaceAll("");
    }
    public static void main(String args[]) {

        // ATOMIC_FUNCTION:  trim($val)
        // LANG:  java
        // FUNCTION: trim($val)

        String start_val = "  abcdefg  ";
        String expected_end_val = "  abcdefg";
        String end_val = rtrim(start_val);



        // TEST
        if(end_val.equals(expected_end_val)){
            System.out.println("PASS");
        } else {
            System.out.println("FAIL");
        }
    }
}
